api/src/test/resources/io/strimzi/api/kafka/model/KafkaClusterRebalance-with-goals-verbose.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaClusterRebalance
+metadata:
+  name: my-rebalance
+spec:
+  goals:
+    - DiskCapacityGoal
```

Comment:
So these goals have to be pre-installed in CC. Do we know what happens if CC gets a bad/not-installed goal name? Does it throw an error? Should we proporgate that back to the user in the CR status?

Reply:
This is probably a job for later but I thought I should flag it

Reply:
it was asked by @tombentley as well on the other PR ... let me report back here my answer.

You can find the list of goals here https://github.com/linkedin/cruise-control#goals. It's possible for a CC user writing its own goal library but it's something that we are not going to support right now. It should be something possible like we do for Kafka Connect: the user writes the goal, build a jar and should include it into a custom Docker image to use as image for the cruise control. Tbh I am fine on listing some of the goals in the description but I would not add any constraint on the CRD itself. If the user set not existing goals in the list, the operator will send the request to CC anyway but will get an error which will be reported in the status of the custom resource.